src/core/cpu_mesh.rs

```diff
@@ -80,6 +80,94 @@ impl std::fmt::Debug for CPUMesh {
     }
 }
 
+///
+/// Specifies the functions used to generate a mesh.
+///
+/// Fields with `None` will generate a mesh without that field.
+///
+pub struct CPUMeshGenerator<'t, P> {
+    /// Generates the position buffer.
+    pub positions: &'t mut (dyn FnMut(P) -> [f32; 3] + 't),
+    /// Generates the normals buffer.
+    pub normals: Option<&'t mut (dyn FnMut(P) -> [f32; 3] + 't)>,
+    /// Generates the tangents buffer.
+    pub tangents: Option<&'t mut (dyn FnMut(P) -> [f32; 4] + 't)>,
+    /// Generates the uv buffer.
+    pub uvs: Option<&'t mut (dyn FnMut(P) -> [f32; 2] + 't)>,
+    /// Generates the color buffer.
+    pub colors: Option<&'t mut (dyn FnMut(P) -> [u8; 4] + 't)>,
```

Comment:
I am not sure if it is OK to use dynamic dispatch here. Another possible way without introducing 5 type parameters in everywhere is to create a trait and use associated types, but that feels like too much work for a simple use case.

I suppose the compiler can inline the dynamic dispatch anyway, and even if it doesn't, users shouldn't call mesh generation functions in hot paths anyway.